src/backend/access/external/hd_work_mgr.c

```diff
@@ -216,51 +216,50 @@ static void assign_pxf_port_to_fragments(int remote_rest_port, List *fragments)
 }
 
 /*
- * Fetches statistics of the PXF datasource from the PXF service
+ * Fetches fragments statistics of the PXF datasource from the PXF service
  *
- * The function will generate a delegation token when secure filesystem mode 
+ * The function will generate a delegation token when secure filesystem mode
  * is on and cancel it right after.
  */
-PxfStatsElem *get_pxf_statistics(char *uri, Relation rel, StringInfo err_msg)
+PxfFragmentStatsElem *get_pxf_fragments_statistics(char *uri, Relation rel, StringInfo err_msg)
```

Comment:
Removed all special error handling. Now if something goes wrong, an error is emitted.
The former behaviour was required for two reasons:
1. ANALYZE on all tables was done as one transaction, but now that we introduced subtransaction per table, it is not needed anymore. 
2. PXF had an implicit ANALYZE run from the planner if no statistics were gathered before. In that case, an error had to be suppressed, which was done by writing the error into a string and ignoring it if necessary. We removed this behaviour as part of this feature, so there is no need to suppress errors anymore.